We don't want our users to update pip themselves, because our pip is patched to
prevent damage to IPS and other Python-related software. Hence, showing them a
message about newer available versions is unnecessary and potentially risky.

Not for upstream contribution

--- pip-19.1.1/src/pip/_internal/utils/outdated.py
+++ pip-19.1.1/src/pip/_internal/utils/outdated.py
@@ -139,22 +139,6 @@ def pip_version_check(session, options):
 
         remote_version = packaging_version.parse(pypi_version)
 
-        # Determine if our pypi_version is older
-        if (pip_version < remote_version and
-                pip_version.base_version != remote_version.base_version and
-                was_installed_by_pip('pip')):
-            # Advise "python -m pip" on Windows to avoid issues
-            # with overwriting pip.exe.
-            if WINDOWS:
-                pip_cmd = "python -m pip"
-            else:
-                pip_cmd = "pip"
-            logger.warning(
-                "You are using pip version %s, however version %s is "
-                "available.\nYou should consider upgrading via the "
-                "'%s install --upgrade pip' command.",
-                pip_version, pypi_version, pip_cmd
-            )
     except Exception:
         logger.debug(
             "There was an error checking the latest version of pip",
